eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,94 @@
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+  $command = Join-Path $ToolPath $ToolCommandName
+  # Test to see if the tool properly installed.
+  if (!(Get-Command $command -errorAction SilentlyContinue)) {
+    Write-Error "The retrieve-codeowners tool is not properly installed. Please check your tool path. $ToolPath"
+    return 
   }
+  return $command
 }
 
-$results = $ownedFolders[$target]
-
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function Get-CodeOwners ([string] $command)
+{
+  if (!$command) {
+    return @()
+  }
+  # Params $RootDirectory is already in use in cpp release pipeline. 
+  # Will use $CodeOwnerFileLocation and deprecate $RootDirectory once it is ready to retire $RootDirectory.
+  if ($RootDirectory -and !(Test-Path $CodeOwnerFileLocation)) {
+    $CodeOwnerFileLocation = Join-Path $RootDirectory ".github/CODEOWNERS"
+  }
+  
+  $codeOwnersString = & $command --target-directory $targetDirectory --code-owner-file-path $CodeOwnerFileLocation
+  # Failed at the command of fetching code owners.
+  if ($LASTEXITCODE -ne 0) {
+    return @()
+  }
+  
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $targetDirectory"
+    return @()
+  }
   
   if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
+    $codeOwners = $codeOwnersJson.Owners -join ","
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
+  }
+
+  return $codeOwnersJson.Owners
+}
+
 
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
+function TestGetCodeOwner([string] $command) {
+  if (!$command) {
+    exit 1
+  }
+  $actualReturn = Get-CodeOwners -command $command 
+  $expectReturn = @("person1", "person2")
+  if ($actualReturn.Length -ne $expectReturn.Length) {
+    Write-Error "The length of actual result is not as expected. Expected length: $expectReturn.Length, Actual length: $actualReturn.Length."
+    exit 1
+  }
+  for ($i = 0; $i -lt $expectReturn.Length; $i++) {
+    if ($expectReturn[$i] -ne $actualReturn[$i]) {
+      Write-Error "Expect result $expectReturn[$i] is different than actual result $actualReturn[$i]."
+      exit 1
     }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
   }
+  exit 0
+}
+
+# Install the tool first
+$output = Get-CodeOwnersTool
 
-  return $results
+if($Test) {
+  TestGetCodeOwner -command $output
 }
 else {
-  Write-Host "Unable to match path $target in CODEOWNERS file located at $codeOwnersLocation."
-  Write-Host ($ownedFolders | ConvertTo-Json)
-  return ""
+  return GetCodeOwners -command $output
```

Comment:
Why are we passing the command still? I thought we were going to move it into the function call itself.